aten/src/ATen/native/UnaryOps.cpp

```diff
@@ -80,6 +80,11 @@ Tensor real(const Tensor& self) {
 
 Tensor imag(const Tensor& self) {
   TORCH_CHECK(!self.is_complex(), "Imag is not yet implemented for complex tensors.");
+  TORCH_CHECK(self.is_complex(),
```

Comment:
should we just...delete it?  It's kind of mean to tell people it must be a complex tensor but then if they pass a complex tensor that it's not implemented.  Sometimes that's unavoidable (e.g. if the errors are far apart), but it seems a bit silly here.

Alternatively, write a global error message?  Just something like "imag is not currently supported".